app/models/concerns/hostext/search.rb

```diff
@@ -109,11 +109,17 @@ def search_by_user(key, operator, value)
         end
         key_name = User.connection.quote_column_name(clean_key)
         condition = sanitize_sql_for_conditions(["#{key_name} #{operator} ?", value_to_sql(operator, value)])
-        users = User.where(condition)
-        hosts = users.map(&:hosts).flatten
-        opts  = hosts.empty? ? "< 0" : "IN (#{hosts.map(&:id).join(',')})"
 
-        {:conditions => " hosts.id #{opts} " }
+        user_ids = User.where(condition).reorder(nil).distinct.pluck(:id)
+        usergroup_ids = Usergroup.joins(:cached_usergroup_members).where(cached_usergroup_members: {user_id: user_ids}).reorder(nil).distinct.pluck(:id)
```

Comment:
This query is unnecessary if the user_ids array is empty and should be not performed to improve performance in this edge case.